src/NewProject/add_constraints_form.cpp

```diff
@@ -22,6 +26,13 @@ addConstraintsForm::addConstraintsForm(QWidget *parent)
 
   ui->m_ckkBoxCopy->setText(tr("Copy sources into project."));
   ui->m_ckkBoxCopy->setCheckState(Qt::CheckState::Checked);
+  ui->select_defineOrder->setChecked(true);
+  Compiler *compiler = GlobalSession->GetCompiler();
+  compiler->PinAssignOpts(Compiler::PinAssignOpt::In_Define_Order);
```

Comment:
shouldn't be here. But you need set ui->select_defineOrder or i->select_random depends on compiler->PinAssignOpts()